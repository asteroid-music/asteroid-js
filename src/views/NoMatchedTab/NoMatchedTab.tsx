//Import from external 'react' module
import React from 'react';

//Import from external '@material-ui/core' module
import Typography from '@material-ui/core/Typography';
import Container from '@material-ui/core/Container';

//Import from external 'react-router-dom' module
import { useLocation } from 'react-router-dom';

/**
 * React component for the main body of the page with no Matched tab
 */
function NoMatchedTab(props: {}) {
    let location = useLocation();
    return <Container>
        <Typography>
            404: No page <code>{location.pathname}</code> found!
        </Typography>
    </Container>
}

export default NoMatchedTab;
