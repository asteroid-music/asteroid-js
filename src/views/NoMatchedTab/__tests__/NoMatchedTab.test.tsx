import React from 'react';
import renderer from 'react-test-renderer';
import NoMatchedTab from 'views/NoMatchedTab';
import { MemoryRouter } from 'react-router-dom';

describe('<NoMatchedTab />',()=>{
    //Basic rendering with minimal props
    describe('basic rendering',()=>{
        it("renders correctly",()=>{
            expect(renderer
                .create(<MemoryRouter initialEntries={["/IncorrectAddress"]}><NoMatchedTab/></MemoryRouter>)
                .toJSON()
            ).toMatchSnapshot();
        });
    });
});