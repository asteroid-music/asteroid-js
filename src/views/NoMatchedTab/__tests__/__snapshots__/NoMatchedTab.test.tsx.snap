// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoMatchedTab /> basic rendering renders correctly 1`] = `
<Material UI Container>
  <Material UI Typography>
    404: No page 
    <code>
      /IncorrectAddress
    </code>
     found!
  </Material UI Typography>
</Material UI Container>
`;
